proposals/proposals.py

```diff
@@ -109,53 +59,92 @@ async def prepare_draft(
 
         return id_type, channel_name, title
 
+    @staticmethod
+    async def get_forum_channel(
+        draft_type: str, bot: commands.Bot, guild_id: int
+    ) -> discord.ForumChannel:
+        """Get the appropriate forum channel based on draft type"""
+        guild = bot.get_guild(guild_id)
+        if not guild:
+            logger.error(f"Guild not found: {guild_id}")
+            return None
+
+        channel_name = (
```

Comment:
might be better to throw an error if there is not channel match, rather than just an ternary that falls back to governance channel